x/evm/types/msg.go

```diff
@@ -382,6 +415,32 @@ func (msg *MsgEthereumTx) BuildTx(b client.TxBuilder, evmDenom string) (signing.
 	return tx, nil
 }
 
+// TxAsMessage ia a modified version of ethtypes.AsMessage that allows to set from address explicitly.
+func TxAsMessage(tx *ethtypes.Transaction, baseFee *big.Int, from common.Address) ethtypes.Message {
```

Comment:
it does the same as `MsgEthereumTx` but does not derive the sender address from the signer and allows to specify from explicitly.